src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java

```diff
@@ -0,0 +1,83 @@
+package com.based.basedsurvey.controller;
+
+import com.based.basedsurvey.data.MultiplechoiceQuestion;
+import com.based.basedsurvey.data.Question;
+import com.based.basedsurvey.data.QuestionTypes;
+import com.based.basedsurvey.data.Survey;
+import com.based.basedsurvey.repo.QuestionRepository;
+import com.based.basedsurvey.repo.SurveyRepository;
+import lombok.extern.java.Log;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+@Log
+@Controller
+public class WebEditSurveyController {
+
+    SurveyRepository surveyRepository;
+    QuestionRepository questionRepository;
```

Comment:
Maybe make surveyRepository and questionRepository private.

Reply:
done! :)